src/cartesian/ticks/utils.ts

```diff
@@ -0,0 +1,59 @@
+import { CartesianTickItem } from '../../util/types';
+
+/**
+ * Given an array and a number N, return a new array which contains every nTh
+ * element of the input array. For n below 1, an empty array is returned.
+ * @param {any[]} array An input array.
+ * @param {integer} n A number
+ * @returns {any[]} The result array of the same type as the input array.
+ */
+function getEveryNth<Type>(array: Type[], n: number): Type[] {
+  if (n < 1) {
+    return [];
+  }
+  if (n === 1) {
+    return array;
+  }
+  const result = [];
+  for (let i = 0; i < array.length; i += n) {
+    result.push(array[i]);
+  }
+  return result;
+}
+
+/**
+ * Given an array of ticks, find N, the lowest possible number for which every
+ * nTH tick in the ticks array isShow == true and return the array of every nTh tick.
+ * @param {CartesianTickItem[]} ticks An array of CartesianTickItem with the
+ * information whether they can be shown without overlapping with their neighbour isShow.
+ * @returns {CartesianTickItem[]} Every nTh tick in an array.
+ */
+export function getEveryNThTick(ticks: CartesianTickItem[]) {
+  let N = 1;
+  let previous: CartesianTickItem[] = ticks;
+  while (N < ticks.length) {
+    const everyNthTick = getEveryNth(ticks, N);
+    const visibleTicks = everyNthTick.reduce((previousValue, currentValue) => {
+      if (currentValue.isShow === true) {
```

Comment:
nit: same as if (currentValue.isShow) {

Reply:
I do not like to rely on truthy and falsy, because they are partially surprising / inconsistent. 
0, for example is falsy, 1 is truthy. In this case only true is valid. 
Thus, I would always nit the the other way around :D

Reply:
<img width="994" alt="Screen Shot 2023-02-23 at 6 28 54 AM" src="https://user-images.githubusercontent.com/60944077/220905863-49b79080-c6e4-4722-866f-4d3a54f56b99.png">

`isShow` is an optional `boolean`, so via TypeScript we know that it can only ever be `true`, `false`, or `undefined`, so as the value would never be `0`. If we were worried about `undefined` being conflated with `false` that would be one thing, but we're not in this case, as we're checking truthiness and not falsyness so @ckifer's suggestion seems pretty safe here.

Reply:
@peterklingelhofer Fair enough, in this specific case that is true. Yet with using any, Typescript is muted, and my reservations against truthy and falsy are general. I prefer to avoid it because that preference has in edge cases prevented bugs. 

Yet this is not going to be the hill to die on for me, happy to change this here. :D

Reply:
No worries - it's personal preference. Agreed with your reservations though also agreed with the counter above. Not the hill anyone should die on. Being explicit is good